runtime/service/npud/core/DBus.cc

```diff
@@ -155,5 +156,83 @@ gboolean DBus::on_handle_request_destroy(NpudCore *object, GDBusMethodInvocation
   return TRUE;
 }
 
+gboolean DBus::on_handle_request_set_data(NpudCore *object, GDBusMethodInvocation *invocation,
+                                          guint64 arg_ctx, guint arg_rq_handle,
+                                          GVariant *arg_input_buffers, GVariant *arg_output_buffers)
+{
+  VERBOSE(DBus) << "on_handle_request_set_data with " << arg_ctx << ", " << arg_rq_handle
+                << std::endl;
+  GVariantIter *iter = NULL;
+  InputBuffers inBufs;
+  OutputBuffers outBufs;
+  gint32 type;
+  guint64 addr;
+  guint32 size;
+  int index = 0;
+
+  // inBufs
+  g_variant_get(arg_input_buffers, "a(itu)", &iter);
+  index = 0;
+  while (iter != NULL && g_variant_iter_loop(iter, "(itu)", &type, &addr, &size))
+  {
+    VERBOSE(DBus) << "in [" << index << "] Type: " << type << ", Addr: " << (void *)addr
+                  << ", Size: " << size << std::endl;
+    if (type == 0) // NPU_BUFFER_MAPPED
```

Comment:
@jyoungyun Why do you use magic number? You've defined the enum in `npud/core/Backend.h`.

Reply:
Um, the `Backend.h` file defines things related to `Backend`. So I try to use the file only between the `DevManager` that calls the `Backend` and the `Backend` library implementation.
This file is DBus file. This is responsible for invoking the appropriate npud behavior by receiving the dbus request by the client.
Thus, these values will be a protocol between client and npud that is different from `Backend.h` file.
When the npud library is created in the future, a header file for the library will be created and these contents will be defined and put in it.

Reply:
@jyoungyun 

1. What is `core/Backend.h` for? Is it something for common part of several backend implementation ? `DBus.h/cc` seems to dependent to `Backend.h` though it is not explicitly shown because it is hidden by `GVariant`. 

2. I don't fully understand what you mean by followings:

> When the npud library is created in the future, a header file for the library will be created and these contents will be defined and put in it.

Do you mean `enum BufferTypes` will be a part of header in the future npud library's header?

Reply:
> 1. What is `core/Backend.h` for? Is it something for common part of several backend implementation ? `DBus.h/cc` seems to dependent to `Backend.h` though it is not explicitly shown because it is hidden by `GVariant`. 

The `core/Backend.h` is a protocol between npud dev manager and npud backend library.
Now, the two (Backend.h and DBus.h) seem to be closely related, but the role is different. There are some similar things because the data received from the user will pass the npu engine. However, the roles and layers are different from each other.
So I think they should be designed separately.

> Do you mean `enum BufferTypes` will be a part of header in the future npud library's header?

Yes. It will be.

Reply:
@jyoungyun `Backend` and `DBus` seems tightly-coupled though you says it has different roles. Also, I don't have a picture how this dependency will be solved in this future. Anyway, I understood you're going to fix it later.